collections/mod.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+/*
+ * Stack is a first-in-last-out data structure.
+ *
+ * Create by optionally passing an Iterable:
+ *
+ *      const s = Stack([1, 2, 3])
+ *      console.log(s.pop())  // 3
+ *
+ */
+export class Stack<T> implements Iterable<T> {
+  private _array: T[];
+  constructor(ls?: Iterable<T>) {
+    this.clear();
+    for (const item of ls || []) {
+      this._array.push(item);
+    }
```

Comment:
this isn't O(1)

Reply:
Really? I thought it was amortized O(1) https://stackoverflow.com/a/11535121/1240268

Reply:
Oh, construction. No, construction is not.

Reply:
Just spread... `this._array = ls ? [...ls] : []`;